hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/BlockRetentionManager.java

```diff
@@ -0,0 +1,170 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * The BlockRetentionManager is responsible for managing the lifecycle of block files in the uploaded directory after a configurable retention period.
+ */
+@Singleton
+public class BlockRetentionManager {
+    private static final Logger log = LogManager.getLogger(BlockRetentionManager.class);
+    private final Path uploadedDir;
+    private final Duration retentionPeriod;
+    private final Duration cleanupInterval;
+    private final ScheduledExecutorService scheduler;
+    private final ExecutorService cleanupExecutor;
+    public static final String BLOCK_FILE_EXTENSION = ".blk";
+    public static final String BLOCK_FILE_EXTENSION_GZ = ".blk.gz";
+
+    /**
+     * Creates a new BlockRetentionManager instance.
+     * @param uploadedDir the directory where uploaded block files are stored
+     * @param retentionPeriod the duration after which block files are considered expired and eligible for deletion
+     * @param cleanupInterval the interval at which the cleanup task should run
+     * @param cleanupThreadPoolSize the number of threads to use for cleanup tasks
+     */
+    @Inject
+    public BlockRetentionManager(
+            @NonNull final Path uploadedDir,
+            @NonNull final Duration retentionPeriod,
+            @NonNull final Duration cleanupInterval,
+            final int cleanupThreadPoolSize) {
+        this.uploadedDir = requireNonNull(uploadedDir, "uploadedDir must not be null");
+        this.retentionPeriod = requireNonNull(retentionPeriod, "retentionPeriod must not be null");
+        this.cleanupInterval = requireNonNull(cleanupInterval, "cleanupInterval must not be null");
+
+        this.cleanupExecutor = Executors.newFixedThreadPool(cleanupThreadPoolSize);
+        this.scheduler = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    /**
+     * Schedules a cleanup task to run at a fixed interval.
+     *
+     * @param cleanupTask the task to run
+     */
+    public void scheduleRepeating(@NonNull final Runnable cleanupTask) {
+        requireNonNull(cleanupTask, "cleanupTask must not be null");
+        scheduler.scheduleAtFixedRate(
+                cleanupTask, cleanupInterval.toMillis(), cleanupInterval.toMillis(), TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Starts the cleanup process asynchronously.
+     */
+    public void startCleanup() {
+        CompletableFuture.runAsync(this::cleanupExpiredBlocks, cleanupExecutor).exceptionally(ex -> {
+            log.warn("Error during cleanup: {}", ex.getMessage());
+            return null;
+        });
+    }
+
+    /**
+     * Deletes all block files that have expired.
+     */
+    public void cleanupExpiredBlocks() {
```

Comment:
I wonder if it makes sense to have a timeout or batch mechanism for this task, e.g. if there are so many block files already on disk that cleaning them _all_ up takes significant processing. It's probably not necessary to consider until we see a problem, but it may be worth keeping in mind